app/Services/RestClient.php

```diff
@@ -60,16 +60,28 @@ public function __construct(Client $client, ServiceRegistryContract $services, R
      */
     private function injectHeaders(Request $request)
     {
-        $this->setHeaders(
-            [
-                'X-User' => $request->user()->id ?? self::USER_ID_ANONYMOUS,
-                'X-Token-Scopes' => $request->user() && ! empty($request->user()->token()) ? implode(',', $request->user()->token()->scopes) : '',
-                'X-Client-Ip' => $request->getClientIp(),
-                'User-Agent' => $request->header('User-Agent'),
-                'Content-Type' => 'application/json',
-                'Accept' => 'application/json'
-            ]
-        );
+        // Set the default headers
+        $headers = [
+            'X-User' => $request->user()->id ?? self::USER_ID_ANONYMOUS,
+            'X-Token-Scopes' => $request->user() && ! empty($request->user()->token()) ? implode(',', $request->user()->token()->scopes) : '',
+            'X-Client-Ip' => $request->getClientIp(),
+            'User-Agent' => $request->header('User-Agent'),
+            'Content-Type' => 'application/json',
+            'Accept' => 'application/json'
+        ];
+
+        // Check if there are whitelisted custom headers
+        $whiteList = env('X_HEADER_WHITELIST', '');
+
+        if ($whiteList != '') {
+            $whiteList = json_decode($whiteList, true);
+
+            foreach ($whiteList as $key => $value) {
+                $headers[$key] = $value;
+            }
```

Comment:
hmm if I understand correctly, this is not really a white list (a list of allowed headers to pass through) but a pre-defined list of headers to pass? then it should be called something else, eg extra headers

Reply:
Ok...you're right. I think then we should go with the suggestion of @MrDarkSkil

Reply:
I have updated the PR to match the real whitelist functionality.